src/common/temporary_file.cpp

```diff
@@ -0,0 +1,49 @@
+#include "common/temporary_file.hpp"
+
+namespace mesos {
+namespace criteo {
+
+/*
+ * Represent a temporary file that can be either written or read from.
+ */
+TemporaryFile::TemporaryFile() {
+  char filepath[] = "/tmp/criteo-mesos-XXXXXX";
+  int fd = mkstemp(filepath);
+  if (fd == -1)
+    throw std::runtime_error(
+        "Unable to create temporary file to run commands");
```

Comment:
You can use the errno message to provide a more meaningful debug information
https://en.cppreference.com/w/cpp/string/byte/strerror